INGRID/geometry.py

```diff
@@ -976,43 +1012,257 @@ def find_nearest(array, value):
                 u = _poloidal_f(j / (np_lines - 1))
                 _r = splev(u, self.radial_spl[i])
                 Pt = Point((float(_r[0]), float(_r[1])))
-                if self.distortion_correction['active'] and j > 0 and j < np_lines - 1:
-                    Res = self.distortion_correction['resolution']
-                    ThetaMin = self.distortion_correction['theta_min']
-                    ThetaMax = self.distortion_correction['theta_max']
-                    umin = _poloidal_f((j - 1) / (np_lines - 1))
-                    umax = _poloidal_f((j + 1) / (np_lines - 1))
-                    Pt1 = radial_vertices[i][j]
-                    Pt2 = radial_vertices[i][j - 1]
-                    Tag = '---- Correcting points: {},{}'.format(i, j)
-                    Pt = CorrectDistortion(u, Pt, Pt1, Pt2, self.radial_spl[i], ThetaMin, ThetaMax, umin, umax, Res, visual, Tag, verbose)
+                # if self.distortion_correction['active'] and j > 0 and j < np_lines - 1:
+                #     Res = self.distortion_correction['resolution']
+                #     ThetaMin = self.distortion_correction['theta_min']
+                #     ThetaMax = self.distortion_correction['theta_max']
+                #     umin = _poloidal_f((j - 1) / (np_lines - 1))
+                #     umax = _poloidal_f((j + 1) / (np_lines - 1))
+                #     Pt1 = radial_vertices[i][j]
+                #     Pt2 = radial_vertices[i][j - 1]
+                #     Tag = '---- Correcting points: {},{}'.format(i, j)
+                #     Pt = CorrectDistortion(
+                #         u, Pt, Pt1, Pt2, self.radial_spl[i], ThetaMin, ThetaMax, umin, umax, Res, visual, Tag, verbose)
 
                 vertex_list.append(Pt)
+
                 if visual:
                     for p in vertex_list:
-                        plt.plot(p.x, p.y, '.', color='black', markersize=8)
+                        plt.plot(p.x, p.y, '.', color='black',
+                                 markersize=2, marker='+')
 
             radial_vertices.append(vertex_list)
+        radial_vertices.append(self.S_vertices)
+
+        # correction of distortion ----------------------
+
+        if not self.distortion_correction['reverse']:
+            print("distortion correction : regular", "has_limit",
+                  self.distortion_correction['has_limit'],
+                  self.distortion_correction["du_min"], self.distortion_correction[
+                      "MaxTol"], self.distortion_correction["reverse_j"]
```

Comment:
What is the new `reverse` mode for `distortion_correction`? I'm curious about the grid differences. Can you describe the differences so we can document this change?

Reply:
So the idea is to generate the poloidal points starting from the separatrix for all patches to avoid the deformation of the cells near the separatrix (this is because poloidal points are generated from the splines with a normalized distance between 0 and 1). du_min is here to prevent cells form being too close from the patch boundary.